Sources/Core/Microsoft.StreamProcessing/Aggregates/SlidingMaxAggregate.cs

```
@@ -44,51 +52,90 @@ private MinMaxState<T> Accumulate(MinMaxState<T> state, long timestamp, T input)
             }
             else
             {
-                if (state.currentTimestamp != InvalidSyncTime)
-                    state.savedValues.Add(state.currentValue);
+                // save only if new input is smaller, otherwise overwrite current with new
+                if (state.currentTimestamp != InvalidSyncTime &&
+                    this.comparer(input, state.currentValue) < 0)
+                {
+                    PushToCollection(state);
+                }
                 state.currentTimestamp = timestamp;
                 state.currentValue = input;
             }
             return state;
         }
 
+        private void PushToCollection(MinMaxState<T> state)
+        {
+            Contract.Assert(state.currentTimestamp != InvalidSyncTime);
+            Contract.Assert(state.values.Count == 0 || state.values.Last().timestamp <= state.currentTimestamp); // Ensure new timestamp is higher
+
+            int index = state.values.FindLastIndex(e => this.comparer(e.value, state.currentValue) > 0);
+            if (index == -1)
+                state.values.Clear();
+            else
+                state.values.RemoveRange(index + 1, state.values.Count - index - 1);
+
+            state.values.Add(new ValueAndTimestamp<T> { timestamp = state.currentTimestamp, value = state.currentValue });
+            state.currentTimestamp = InvalidSyncTime;
+        }
+
         public Expression<Func<MinMaxState<T>, long, T, MinMaxState<T>>> Deaccumulate()
             => (state, timestamp, input) => state; // never invoked, hence not implemented
```

>=> (state, timestamp, input) => state; // never invoked, hence not implemented [](start = 12, length = 78)

Maybe we should throw an exception here to be sure